File: packages/@goodauldbooks/library/src/domain.ts
export const LANG_ALL = "_all_";

export enum BookAssetType {
  COVER = "cover",
  EPUB = "epub",
  MOBI = "mobi",
  TXT = "txt",
}

export interface BookAsset {
  type: BookAssetType;
  size: number;
}

export interface Author {
  provider: string;
  id: string;
  firstName: string | null;
  lastName: string | null;
  birthYear: number | null;
  deathYear: number | null;
}

export interface Book {
  provider: string;
  id: string;
  title: string;
  lang: string;
  genres: string[];
  assets: BookAsset[];
  authors: Author[];
}

File: src/app/library/domain.py
import typing as t
from enum import Enum
from dataclasses import dataclass

LANG_ALL = "_all_"


class BookAssetType(Enum):
    EPUB = "epub"
    MOBI = "mobi"


@dataclass
class BookAsset:
    type: BookAssetType
    size: int


@dataclass
class Author:
    provider: str
    id: str
    name: str
    first_name: str = None
    last_name: str = None
    birth_year: int = None
    death_year: int = None


@dataclass
class Book:
    provider: str
    id: str
    title: str
    subtitle: t.Optional[str]
    lang: str
    genres: t.List[str]
    assets: t.List[BookAsset]
    authors: t.List[Author]
    intro: t.Optional[str]
